### PR TITLE
CardCheckbox and CardRadioGroup Components

### PR DESCRIPTION
**Pull request by coderitual**
CardCheckbox and CardRadioGroup.

CleanShot 2023-05-19 at 10 10 17
CleanShot 2023-05-19 at 10 10 38
CleanShot 2023-05-19 at 10 10 55
CleanShot 2023-05-19 at 10 11 05
CleanShot 2023-05-19 at 10 11 15

### CODE REVIEW

File: src/components/card-interactive/CardRadio.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  /**
+   * Optional string. Additional class names.
+   */
+  className?: string;
+
+  /**
+   * Optional React.ReactNode. Children of the card. This is the place where label should be used and connected to the card.
+   * @example <CardRadio>Card content</CardRadio>
+   */
+  children?: React.ReactNode;
```
Comment (clxandstuff) [inline src/components/card-interactive/CardRadio.tsx:57]: You don't need to type `children` when using `React.forwardRef`

File: src/components/card-interactive/CardRadio.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+  /**
+   * Optional string. Width of the card.
+   * @default auto
```
Comment (clxandstuff) [inline src/components/card-interactive/CardRadio.tsx:61]: This directive describes JS prop, but the default value is set in only in CSS, so maybe we should remove it?

File: src/components/card-interactive/CardRadio.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+   * @example <CardRadio width="100px" />
+   */
+  width?: React.CSSProperties['width'];
+
+  /**
+   * Optional string. Height of the card.
+   * @default auto
```
Comment (clxandstuff) [inline src/components/card-interactive/CardRadio.tsx:68]: This directive describes JS prop, but the default value is set in only in CSS, so maybe we should remove it?

File: src/components/card-interactive/CardRadioGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+const CardRadioGroupRoot = React.forwardRef<
+  HTMLDivElement,
+  CardRadioGroupPropsType
+>((props, ref) => {
+  const {
+    name,
+    required = false,
+    disabled = false,
+    invalid,
```
Comment (clxandstuff) [inline src/components/card-interactive/CardRadioGroup.tsx:86]: According to JSDoc comments it should be `false` by default

```suggestion
    invalid = false,
```

**Review by clxandstuff — CHANGES_REQUESTED** [review]

File: src/components/card-interactive/CardCheckbox.tsx [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+  /**
+   * Optional string. Additional class names.
+   */
+  className?: string;
+
+  /**
+   * Optional React.ReactNode. Children of the card. This is the place where label should be used and connected to the card.
+   * @example <CardCheckbox>Card content</Card.Checkbox>
+   */
+  children?: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (coderitual) [reply src/components/card-interactive/CardCheckbox.tsx:24]: why is that?

[the remark this answers is not in the document]
Comment (coderitual) [reply src/components/card-interactive/CardCheckbox.tsx:24]: pls take a look here. the type in forwardRef typescript definition is wrong:

https://github.com/facebook/react/blob/5309f102854475030fb91ab732141411b49c1126/packages/react/src/ReactForwardRef.js#L12

**Review by clxandstuff — APPROVED** [review]

File: src/components/card-interactive/CardCheckbox.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  it('renders as unchecked by default', () => {
+    const checkbox = render(
+      <CardCheckbox defaultChecked={false}>my label</CardCheckbox>
+    );
+    const checkboxInput = checkbox.getByRole('checkbox') as HTMLInputElement;
+
+    expect(checkboxInput.checked).toBe(false);
+  });
+
+  it('works in controlled mode', () => {
```
[the remark this answers is not in the document]
Comment (coderitual) [reply src/components/card-interactive/CardCheckbox.spec.tsx]: Done

File: src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  {
+    pattern: '<b>Should</b> have visible focus and hover style.',
+    status: 'DONE',
+    tests: 'N/A',
+  },
+  {
+    pattern:
+      '<b>Should</b> have a label with 4.5:1 contrast ratio against the background.',
+    comment:
+      '<code>dark</code> against white: 21:1, <code>light</code> against black: 20.9:1.',
```
Comment (katarzynatobis) [inline src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts:77]: Label color is not set by default, it depends on the content. 

```
comment: 'Selected <code>color</code> should relate to the label color.'
```

File: src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    status: 'DONE',
+    tests: 'N/A',
+  },
+  {
+    pattern:
+      '<b>Should</b> have a color indicator with 3:1 contrast ratio against the background.',
+    comment:
+      '<code>dark</code> against white: 21:1, <code>light</code> against black: 20.9:1.',
+    status: 'DONE',
```
Comment (katarzynatobis) [inline src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts:86]: light against black is 13.48:1

File: src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+  {
+    pattern: '<b>Should</b> respect Windows High Contrast mode.',
+    status: 'TO DO',
+    tests: 'N/A',
+  },
+  {
+    pattern:
+      '<b>Should</b> have descriptive information about required and invalid state.',
+    comment:
+      'Invalid state is indicated by color change. Both states can be described using <code>description</code> prop.',
```
Comment (katarzynatobis) [inline src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts:98]: 'Invalid state is indicated by color change. Both states can be described using <code>**aria-describedby**</code> prop.'

File: src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+  {
+    pattern: '<b>Can</b> have a clickable label.',
+    comment: 'The whole area of card is wrapped with clickable label.',
+    status: 'DONE',
+    tests: 'DONE',
+  },
+  {
+    pattern: '<b>Can</b> have an accessible description.',
+    comment:
+      'Can be described by setting a value for <code>description</code> prop.',
```
Comment (katarzynatobis) [inline src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts:127]: aria-description, not description :)

**Review by katarzynatobis — CHANGES_REQUESTED** [review]

File: src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts (hunk 8 above)
Comment (coderitual) [reply src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts:86]: Done

File: src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts (hunk 9 above)
Comment (coderitual) [reply src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts]: Done

File: src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    tests: 'DONE',
+  },
+  {
+    pattern: '<b>Should</b> have only one checked radio at a time.',
+    status: 'DONE',
+    tests: 'DONE',
+  },
+  {
+    pattern: '<b>Should</b> have descriptive information about invalid state.',
+    comment: 'State can be described using <code>errorMessage</code> prop.',
```
[the remark this answers is not in the document]
Comment (coderitual) [reply src/components/card-interactive/stories/CardRadioGroup.rules.a11y.ts]: Done

File: src/components/card-interactive/stories/CardRadioGroup.a11y.mdx [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import {rules, radioItemRules} from './CardRadioGroup.rules.a11y';
+
+### Rules
+
+#### `<CardRadioGroup>`
```
[the remark this answers is not in the document]
Comment (coderitual) [reply src/components/card-interactive/stories/CardRadioGroup.a11y.mdx]: Done

**Review by katarzynatobis — APPROVED** [review]

**Pull request merged**